### DISCUSSION TITLE
Allow FAIL_ON_UNKNOWN_PROPERTIES to be enabed/disabled via property

### DISCUSSION THREAD

**Discussion by Andy-Turner** [openrewrite/rewrite#4375] [Ideas] [answered]
I was debugging an edge case we saw in one of our recipes and it was due to using `artifactId` rather than `artifactPattern` option in a recipe. This was still a valid recipe as `artifactPattern` is optional and the `JsonMapper` has `disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES)` configured - so `artifactId` was ignored and `artifactPattern` was null which then matched *.

I think would be nice to be able to set recipes to fail on unknown properties so that cases such as this could be more easily caught during testing - unless there is already such functionality I'm not aware of. E.g. the following look very similar during a code review:

Incorrect:
```
  - org.openrewrite.maven.RemoveRedundantDependencyVersions:
      groupId: org.apache.commons
```

Correct:
```
  - org.openrewrite.maven.RemoveRedundantDependencyVersions:
      groupPattern: org.apache.commons
```

**Comment by timtebeek** [comment]
Got to say I heavily agree; would welcome any help towards this!

**Reply by Andy-Turner** [reply]
👍 I can probably attempt this sometime in the next 2-3 weeks.

**Reply by timtebeek** [reply]
Much appreciated! Indeed would help prevent confusion and time lost to unknown options.

**Reply by ajturnerora** [reply, marked as answer]
- PR: https://github.com/openrewrite/rewrite/pull/4377
